### ISSUE TITLE
[$250] mWeb - When tapping back, the header on mWeb stays purple for a couple of seconds

### ISSUE THREAD

**Issue by mitarachim** [Expensify/App#54602]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.79-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://github.com/Expensify/App/pull/53686
**Email or phone of affected tester (no customers):** applausetester+en@applause.expensifail.com
**Issue reported by:** Applause Internal Team
**Device used:** iPhone 15 Pro / 18
**App Component:** Chat Report View

## Action Performed:
1. Access staging.new.expensify.com
2. Open FAB > Submit expense.
3. Enter amount > Next.
4. Tap "Submit an expense, get $250" banner.
5. Tap back
6. Verify that: The device navigation bar will not have purple color.

## Expected Result:
User expects the header to turn back to green immediately after tapping back

## Actual Result:
The top nav menu turns purple for a couple of seconds and stays purple when tapping back, even after exiting the banner page

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![Bug6702727_1735261423200!Purple_header_](https://github.com/user-attachments/assets/d22fcff2-64c9-4869-9ba8-642b8f01b718)

https://github.com/user-attachments/assets/eb3fee83-01bf-49d0-be31-b70c49ecb375

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021873780397844730879</li>
        <li>Upwork Job ID: 1873780397844730879</li>
        <li>Last Price Increase: 2025-01-20</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ahmedGaber93</details>

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When tapping back, the header on mWeb stays purple for a couple of seconds

### What is the root cause of that problem?

We add a delay here to wait 300ms after the transition starts.

https://github.com/Expensify/App/blob/0667cf722d040636d7b1b1570a105e98cae15381/src/components/CustomStatusBarAndBackground/index.tsx#L105

I tested this issue, and it only happens on mWeb, where it stays purple for a couple of seconds. It doesn’t occur on native because of differences in how browsers handle animations, rendering, or event timing compared to native platforms.

### What changes do you think we should make in order to solve the problem?


This PR (https://github.com/Expensify/App/pull/33200/files) adds a delay to the status bar animation.

To resolve this issue, we should remove the delay for mWeb to avoid the delay for mweb.

https://github.com/Expensify/App/blob/0667cf722d040636d7b1b1570a105e98cae15381/src/components/CustomStatusBarAndBackground/index.tsx#L105

update to

```diff
  statusBarAnimation.set(withDelay(Browser.isMobile() ? 0 : 300, withTiming(1)));
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


This is a UI bug and doesn’t require a unit test

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by ahmedGaber93** [comment]
Reviewing today

**Comment by ahmedGaber93** [comment]
While removing the delay can fix the start timing issue, but the animation of status bar and page looks inconsistent, status bar is using fading and page content is using sliding. I think we need to fix that if it is available?

https://github.com/user-attachments/assets/4586dd6b-8f31-489d-9dc9-20100f45e99b

**Comment by ahmedGaber93** [comment]
Keeping it open for a better proposal if available https://github.com/Expensify/App/issues/54602#issuecomment-2571469795

**Comment by ahmedGaber93** [comment]
Not overdue, still looking for proposals https://github.com/Expensify/App/issues/54602#issuecomment-2571469795.

**Comment by ahmedGaber93** [comment]
Asking for proposals on slack https://expensify.slack.com/archives/C01GTK53T8Q/p1737374826065079

**Comment by shawnborton** [comment]
Ah actually I see, that's because this is an mWeb thing... that being said, not sure what we can really do about this, I would be fine just closing personally.

**Comment by dannymcclain** [comment]
Yeah I would fine closing too. Seems like we don't have a ton of options for "fixing" this, and it's also not a super big deal IMO.

**Comment by ahmedGaber93** [comment]
> @ahmedGaber93 There is no way we can set the animation of the bar color. We achieve fading by gradually changing the color value via js, creating a fade effect. The sliding effects are not possible. Browser controls the theme behaviour.

@parasharrajat Thanks for letting me know 🙏
